Source/Particles/LaserParticleContainer.cpp

```diff
@@ -236,15 +237,22 @@ LaserParticleContainer::UpdateContinuousInjectionPosition (Real dt)
 void
 LaserParticleContainer::InitData ()
 {
+    if (!m_enabled) return;
+
     // Call InitData on max level to inject one laser particle per
     // finest cell.
     InitData(maxLevel());
+
+    if(!do_continuous_injection && (TotalNumberOfParticles() == 0)){
+        amrex::Print() << "WARNING: laser antenna is completely out of the simulation box !!!\n";
```

Comment:
Quick question out of curiosity:
Is there a specific reason for not using directly `amrex::Warning` for the warning message here (as well as the one on line 73 above, where `m_e_max` is checked)?

Reply:
We just checked (cc @RemiLehe) and `amrex::Warning` does not print a message in non-Debug mode. Otherwise, it behaves like a `PrintAll()` to `stderr`.

Reply:
Ah OK, thanks. Then I guess we might want to replace some of the `amrex::Warning` calls that we have elsewhere in the code (I counted 13 instances), because it is quite rare to run it in DEBUG mode, and I guess some of those warning messages were probably meant to be written out always...?

Reply:
I think so, I wasn't aware either :D